clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -8639,6 +8729,15 @@ void KernelCallRule::runRule(
     if (DpctGlobalInfo::getInstance().buildLaunchKernelInfo(LaunchKernelCall)) {
       emplaceTransformation(new ReplaceStmt(LaunchKernelCall, true, false, ""));
       removeTrailingSemicolon(LaunchKernelCall, Result);
+      if (const DeclRefExpr *DRE =
+              getAddressedRef(LaunchKernelCall->getArg(0))) {
+        if (const Decl *D = DRE->getReferencedDeclOfCallee()) {
+          if (const FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(D)) {
+            for (const ParmVarDecl *Parm : FD->parameters())
```

Comment:
param's type may different from call arg's type.

Reply:
CUDA just do plain memcpy for the kernel arguments.
So, I think the parameter type is as same as the argument type (ignoring the implicit case).